src/Morphir/IR/Repo.elm

```diff
@@ -124,9 +126,91 @@ parseNewElmModules fileChanges =
     Debug.todo "implement"
 
 
-orderElmModulesByDependency : List ParsedModule -> Result Errors (List ParsedModule)
-orderElmModulesByDependency parsedModules =
-    Debug.todo "implement"
+orderElmModulesByDependency : Repo -> List ParsedModule -> Result Error (List ParsedModule)
+orderElmModulesByDependency repo parsedModules =
+    {- let
+          finalModuleGraph: DAG ModuleName
+          finalModuleGraph = DAG.empty
+
+          foldLeftInitialFunction: ParsedModule -> DAG ModuleName -> Result Error (DAG ModuleName)
+          foldLeftInitialFunction parsedModule =
+             let
+                  -- extract IR moduleName from ParsedModule
+                  fromModuleNameParam = parsedModule |> ParsedModule.moduleName |> toIRModuleName repo.packageName
+
+                  -- extract parsedMod Dependencies + format each moduleName to type ModuleName
+                  moduleDependencies: List ModuleName
+                  moduleDependencies =
+                      ParsedModule.importedModules parsedModule
+                           |> List.map (toIRModuleName repo.packageName)
+
+                  -- insert function to DAG Graph
+
+              in
+              moduleDependencies
+                   |> List.foldl (foldLeftInnerFunction)
+       in
+       parsedModules
+           |> List.foldl (foldLeftInitialFunction) finalGraph
+           |> Result.andThen (\functionAResult ->
+               case functionAResult of
+                   Err err ->
+                       case  err of
+                           _ ->
+                               CycleDetected
+                   Ok _ ->
+                       parsedModules
+           )
+    -}
+    let
+        moduleGraph : DAG ModuleName
+        moduleGraph =
+            DAG.empty
+
+        foldFunction : ParsedModule -> Result Error (DAG ModuleName) -> Result Error (DAG ModuleName)
+        foldFunction parsedModule graph =
+            let
+                moduleName : ModuleName
+                moduleName =
+                    ParsedModule.moduleName parsedModule
+                        |> toIRModuleName repo.packageName
+
+                moduleDependencies : List ModuleName
+                moduleDependencies =
+                    ParsedModule.importedModules parsedModule
+                        |> List.map (\modName -> toIRModuleName repo.packageName modName)
```

Comment:
This should check if the Elm module name starts with the current package name and ignore any imports that don't satisfy that.